horovod/run/common/util/env.py

```diff
@@ -17,8 +17,15 @@
 
 
 # List of regular expressions to ignore environment variables by.
-IGNORE_REGEXES = {'BASH_FUNC_.*\(\)', 'OLDPWD'}
+IGNORE_REGEXES = {re.compile(r'BASH_FUNC_.*\(\)'), re.compile('OLDPWD'),
+                  # ignore strings with any none word chars (e.g. '<')
+                  re.compile(r'[^\w/.:]'),
+                  # Do not overwrite instance id on each instance
+                  re.compile('MICHELANGELO_INSTANCE_ID'),
```

Comment:
Instead of hardcoding these variables, can we put them in a config file? That will allow users outside of Uber to customize them for their environment.